pki/proxy-node-metadata.rb

```diff
@@ -0,0 +1,91 @@
+
+    Dir.mkdir(metadata_env)
+    Dir.chdir(metadata_env) do
+      Dir.mkdir('idps')
+      Dir.chdir('idps') do
+        create_file('proxy-node.yml', YAML.dump(proxy_node_config))
+        puts(YAML.dump(proxy_node_config) + "---")
+      end
+    end
+
+    generated_xml = capture_output do
+      Verify::Metadata::Generator.run!(%W(-e #{metadata_env} --valid-until=365 --proxy-node --idpCA ca.crt))
+    end
+
+    Nokogiri::XML(generated_xml).tap do |xml|
+      extensions = <<~EOS
+        <md:Extensions>
+          <mdattr:EntityAttributes xmlns:mdattr="urn:oasis:names:tc:SAML:metadata:attribute">
+            <saml2:Attribute xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion" Name="http://eidas.europa.eu/LoA" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
+              <saml2:AttributeValue>http://eidas.europa.eu/LoA/substantial</saml2:AttributeValue>
+            </saml2:Attribute>
+          </mdattr:EntityAttributes>
+          <alg:DigestMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#sha384"/>
+          <alg:DigestMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmlenc#sha512"/>
+          <alg:DigestMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha512"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#ecdsa-sha256"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-ripemd160"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#ecdsa-sha384"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha384"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1"/>
+          <alg:SigningMethod xmlns:alg="urn:oasis:names:tc:SAML:metadata:algsupport" Algorithm="http://www.w3.org/2001/04/xmldsig-more#ecdsa-sha512"/>
+        </md:Extensions>
+      EOS
+
+      xml.at_xpath("//md:EntityDescriptor//md:IDPSSODescriptor").add_previous_sibling(extensions)
+    end
+  end
+end
+
+#
+#   Sign Metadata locally - requires a keypair and an xmlsectool installation
+#
+def sign_metadata(metadata_xml, keypair, xmlsectool_path = nil)
+  xmlsectool_path ||= ENV.fetch('XMLSECTOOL', 'xmlsectool')
```

Comment:
This check should go into `gen_sign_proxy_node_metadata` when the `xmlsectool_path` is extracted.